wgsl/index.bs

```diff
@@ -5001,27 +5012,41 @@ parameters and return types:
   </xmp>
 </div>
 
-## Function calls TODO ## {#function-calls}
-
-A function call is a statement or expression which invokes a function.
-
-TODO: explain how invocation works: supply operands matching formal parameter types,
-"suspend" execution of the caller, then resume after
-the callee is done (unless discard). Describe return value.
-
-TODO: A function call site is a dynamic context. This matters when discussing the originating
-variable for the dynamic value provided as the operand for a formal argument having pointer type.
-
-The names in the parameter list of a function definition are available for use in the body
-of the function.
-During a particular function evaluation,
-the parameter names denote the values specified to the function call expression or statement
-which initiated the function evaluation;
-the names and values are associated by position.
-
-TODO: define 'formal parameter'.
-
-## Function calls TODO ## {#func-call-semantics}
+## Function calls ## {#function-calls}
+
+A <dfn noexport>function call</dfn> is a statement or expression which invokes a function.
+
+When a function call is executed the following steps occur:
+* Execution of the current function is suspended.
+    All [=function scope=] variables and constants maintain their current values.
+* Storage is allocated to the function scope variable in the called function.
+* Values for the [=formal parameter|formal parameters=] of the called function are determined
+    by matching the function call argument values by position the formal
+    parameters (e.g. the first argument is matched to the first formal
+    parameter).
+* Control is transferred to the first statement in the called function
+* The called function is executed
+* When the called function finishes execution by either executing a return statement or reaching
+    the end of the called function (if the function does not return a value),
+    control is transferred back the calling function and execution is
+    unsuspended.
+    If the called function returns a value, that value is supplied for the
+    value of the function call expression.
+
+The type of each argument must match the type of the matched formal parameter.
+A value must be supplied for each formal parameter that is declared without a
+default value.
```

Comment:
There are no default values.   If there's a missing parameter then you must have selected a different function in the first place.

So I suggest:

"The function call must supply the same number of argument values as there are formal parameters in the called function. Each argument value must have the same type as the corresponding formal parameter, by position."

I think this validation rule should be placed before the behavioural description.  The behaviour doesn't make sense if the rule is violated.